.golangci.yml

```diff
@@ -1,3 +1,6 @@
+run:
+  skip-files:
+    - build/asm_shim/c.go
```

Comment:
If there are lint issues with `build/asm_shim/c.go` should we just fix them rather than skipping the file here?

Reply:
Otherwise, rename `asm_shim` to `asm-shim`, and update `build/go_mock.bzl` to match, testing with `make generate` to ensure nothing else is broken.

Reply:
I'm fine with defering this as long as a tracking issue is created for followup.

Reply:
Tracking at https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/626